packages/concerto-core/lib/decoratormanager.js

```diff
@@ -253,6 +253,233 @@ class DecoratorManager {
         return newModelManager;
     }
 
+    /**
+     * Extracts all the decorator commands from all the models in modelManager
+     * @param {ModelManager} modelManager the input model manager
+     * @param {object} options - decorator models options
+     * @param {boolean} options.removeDecoratorsFromModel - flag to strip out decorators from models
+     * @param {string} options.locale - locale for extracted vocabulary set
+     * @returns {Object} a new model manager with the decorations removed and a list of extracted decorator jsons
+     */
+    static extractDecorators(modelManager,options) {
+        options = {
+            ...{
+                removeDecoratorsFromModel: false,
+                locale:'en'
+            },
+            ...options || {}
+        };
+        const ast = modelManager.getAst(true);
+        const decoratedAst = JSON.parse(JSON.stringify(ast));
+        let extractionDictionary={};
+        let processedModels = decoratedAst.models.map((model)=>{
+            if (model.decorators && model.decorators.length>0){
+                extractionDictionary=this.constructDCSDictionary(extractionDictionary,model.namespace, model.decorators,'','');
+                if (options.removeDecoratorsFromModel){
+                    model.decorators=undefined;
+                }
+            }
+            let processedDecl = model.declarations.map((decl)=>{
+                if (decl.decorators) {
+                    extractionDictionary=this.constructDCSDictionary(extractionDictionary,model.namespace,decl.decorators,decl.name,'');
+                }
+                if (options.removeDecoratorsFromModel){
+                    decl.decorators=undefined;
+                }
+                if (decl.properties) {
+                    let processedProperties = decl.properties.map((property) => {
+                        if (property.decorators){
+                            extractionDictionary=this.constructDCSDictionary(extractionDictionary,model.namespace, property.decorators,decl.name,property.name);
+                        }
+                        if (options.removeDecoratorsFromModel){
+                            property.decorators=undefined;
+                        }
+                        return property;
+                    });
+                    decl.properties=processedProperties;
+                }
+                return decl;
+            });
+            model.declarations=processedDecl;
+            return model;
+        });
+        let processedAST={
+            ...decoratedAst,
+            models:processedModels,
+        };
+        const newModelManager = new ModelManager();
+        newModelManager.fromAst(processedAST);
+        let dcms=(this.parseNonVocabularyDecorators(extractionDictionary));
+        let vocab = this.parseVocabularies(extractionDictionary,options.locale);
```

Comment:
Nit. I'm surprised that the linter didn't pick this up, but we have a preference for `const` declarations unless neccessary in this repo. Also, the whitespace is inconsistent around the assignment.